### DISCUSSION TITLE
External Ceph ObjectStore with multiple StorageClasses / zone placements

### DISCUSSION THREAD

**Discussion by brandonrjacobs** [rook/rook#10431] [General]
Hey everyone, So I'm still fairly new to this ecosystem but have a question as to how rook might handle this or how this should be setup. Currently we run an external Ceph Cluster with a radosgw already setup for use as an objectstore. Our Object store has two zone placement groups, let's call them `default, non-default`.

Each of these groups have a storageClass defined as the `STANDARD` storageClass. 

Given that we plan to create an ObjectStore Resource in Kubernetes and point to our external gateway. How does the creation of StorageClasses work with rook's provisioner for objects?

Ideally we want to define two storageClasses, one for default, and the other for non-default so that a user can create OBC's with the different storage tiers whenever they want. Is this currently possible to achieve? Do we need to place both of these in the same placement but have two StorageClasses instead? 

This is my first time integrating an external cluster like this so I apologize if these questions are vague or not helpful.

**Comment by travisn** [comment]
To connect to the external object store, you can create this [example object store](https://github.com/rook/rook/blob/master/deploy/examples/object-external.yaml) with the endpoints of the external rgw endpoint. Then you can create a storage class for that object store, as if it were a local object store.

I'm not sure about the zone placement groups. Can you point to a ceph doc where that is described? Or maybe @thotz can already answer that.

**Reply by brandonrjacobs** [reply]
Thank you, the external objectstore portion makes sense. As for zone placement groups I'm referring to the documentation here: https://docs.ceph.com/en/quincy/radosgw/placement/#s3-bucket-placement. Ideally we can modify our setup to specify storage class or the placement target but either would be fine it seems. I just am not sure how to integrate this with rook.

**Reply by brandonrjacobs** [reply]
According to the code in the Provisioner, it seems as though it will just use the default placement / storage class when creating a bucket.

**Reply by thotz** [reply]
Yes, all the internal s3 clients used by Rook such as `health checker` or `bucket provisioner` send requests to default(or assumes). For the external rgw server, the option input Rook takes the endpoint and all the requests sent via that endpoint. No other configuration was shared with Rook. If I understand correctly the issue residing with the bucket created via OBC will either end up creating in default placement or it will fail. The hacky way to fix that adds conf options for OBC to tell which placement/storageclass it needs to create/fetch and this will work only with the external rgw server. 
For the internal rgw server, we are still stabilising multisite code and work flow, after we can add support for rgw placement/storage class in the configuration level
